implementation/src/main/java/io/smallrye/graphql/transformation/Transformer.java

```diff
@@ -16,31 +16,49 @@
     UriTransformer URI_TRANSFORMER = new UriTransformer();
 
     static Transformer transformer(Field field) {
-        if (field.hasTransformInfo()) {
-            TransformInfo format = field.getTransformInfo();
-            if (format.getType().equals(TransformInfo.Type.NUMBER)) {
-                if (format.getFormat() != null || format.getLocale() != null) {
-                    return new FormattedNumberTransformer(field);
-                }
-                return new NumberTransformer(field);
+        if (shouldTransform(field)) {
```

Comment:
I would remove this check.  The following if/else if blocks cover all of the conditions that `shouldTransform` checks, so it is essentially checking those conditions twice.